File: retrofit/src/main/java/com/seeyou/retrofit/MainActivity.java
package com.seeyou.retrofit;

import android.os.Bundle;
import android.util.Log;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        new Thread(() -> {

            Retrofit retrofit = new Retrofit.Builder()
                    .baseUrl("http://www.baidu.com/")
                    .addConverterFactory(StringConverterFactory.create())
                    .build();

            BaiduRequest baiduRequest = retrofit.create(BaiduRequest.class);

            BdRequest bdRequest = retrofit.create(BdRequest.class);

            Call<String> bdString = bdRequest.getBdString();
            bdString.enqueue(new Callback<String>() {
                @Override
                public void onResponse(Call<String> call, Response<String> response) {

                }

                @Override
                public void onFailure(Call<String> call, Throwable t) {

                }
            });

            Call<String> baiduString = baiduRequest.getBaiduString();
            baiduString.enqueue(new Callback<String>() {
                @Override
                public void onResponse(Call<String> call, Response<String> response) {
                    Log.e("tsy", "onResponse : " + response.message());
                }

                @Override
                public void onFailure(Call<String> call, Throwable t) {
                    Log.e("tsy", "onFailure");
                }
            });


        }).start();

    }
}

File: okhttp/src/main/java/com/seeyou/okhttp/MainActivity.kt
package com.seeyou.okhttp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import okhttp3.*
import java.io.IOException

class MainActivity : AppCompatActivity() {
    private val TAG = "OkHttp"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        val client = OkHttpClient.Builder().build()

        val request = Request.Builder()
            .url("http://www.baidu.com")
            .build()

        val newCall = client.newCall(request)

        newCall.enqueue(object: Callback {
            override fun onFailure(call: Call, e: IOException) {
                Log.e(TAG, "onFailure")
                e.printStackTrace()
            }

            override fun onResponse(call: Call, response: Response) {
                Log.e(TAG, "onResponse : " + response.message)
            }
        })

    }
}